### DISCUSSION TITLE
Could AppIntro support to lock/unlock pager when navigating by keypad?

### DISCUSSION THREAD

**Discussion by tmshih** [AppIntro/AppIntro#1011] [Ideas]
AppIntroBase support to lock/unlock pager by invoking AppIntroBase.setSwipeLock(Boolean), which applies to the scenario of touch control.

Does AppIntro support to lock/unlock pager by keypad control?
(AppIntroViewPager may override ViewPager.arrowScroll(int) just like https://github.com/tmshih/FragmentPager/blob/master/app/src/main/java/com/tms/fragmentpager/view/SwipeableViewPager.java)

**Comment by cortinico** [comment]
> Does AppIntro support to lock/unlock pager by keypad control?

Could you clarify what would be your expected behavior?

**Reply by tmshih** [reply]
For Android TV or dongle (e.g. https://store.google.com/product/chromecast_google_tv), such devices only provide a remote to interact with various APPs. It's all based on KeyEvent to interact with APP, and the AppIntroBase.setSwipeLock(Boolean) does not apply to lock current page when user press left/right key.

So, AppIntro may provide a way to lock current page when user using a remote (just like AppIntroBase.setSwipeLock(Boolean) could lock the interaction with touch events). It's useful to develop a tutorial for TV APPs.

**Reply by tmshih** [reply]
For example https://github.com/tmshih/FragmentPager/blob/master/app/src/main/java/com/tms/fragmentpager/view/SwipeableViewPager.java 
I'm override the ViewPager.arrowScroll(int) to lock KeyEvent interaction, which may provide a reference.

**Reply by cortinico** [reply]
Yup that makes sense. I believe we can create a `protected fun setArrowScrollLock(lock: Boolean)` function to toggle this 👍 

I'm happy to review a PR to do this if you, or someone else, wish to develop this :)

**Reply by tmshih** [reply]
That's great!
